javascript/modules/client/src/internal/encoding/client.ts

```diff
@@ -26,7 +53,32 @@ function unwrapWithdrawParams(
 
 export function encodeUpdateMetadataAction(ipfsUri: string): Uint8Array {
   const daoInterface = DAO__factory.createInterface();
-  const args = hexToBytes(ipfsUri);
-  const hexBytes = daoInterface.encodeFunctionData("setMetadata", [args])
-  return hexToBytes((strip0x(hexBytes)));
+  const args = new TextEncoder().encode(ipfsUri);
+  const hexBytes = daoInterface.encodeFunctionData("setMetadata", [args]);
```

Comment:
Why do we need to encode the IPFS URI as bytes? Can't it just be the string itself?

Reply:
Nope, the contracts receive an array of bytes